oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexProviderService.java

```diff
@@ -177,7 +180,11 @@ private void deactivate() {
         }
     }
 
-    private void registerIndexCleaner(Config contextConfig) {
+    private void registerIndexCleaner(Config contextConfig) throws IOException {
+        boolean reachable = elasticConnection.getClient().ping(RequestOptions.DEFAULT);
```

Comment:
Can we move this logic inside a method in `ElasticConnection`? Something like 

```
public boolean isAvailable() { ... }
```

Reply:
added.